libnavigation-base/src/main/java/com/mapbox/navigation/base/route/Router.kt

```diff
@@ -1,9 +1,22 @@
 package com.mapbox.navigation.base.route
 
 import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.route.model.Route
 
 interface Router {
 
-    fun getRoute(origin: Point, waypoints: List<Point>, callback: (route: Route) -> Unit)
+    fun getRoute(
```

Comment:
Same as above:

Right now we're not taking `Route` / `RouteOptionsNavigation` (`NavigationRoute`) as a parameter so none of the options from `NavigationRoute` can be configured from a client point of view. Should we request a `Route` / `RouteOptionsNavigation` (`NavigationRoute`) object here (that includes `origin`, `waypoints`, `destination` + other fields) instead of `origin`, `waypoints` and `destination`? That was the purpose of having `NavigationRoute` and we're not using it, we're only setting these 3 parameters and not considering the rest.